docusaurus/docs/how-to-guides/app-plugins/add-resource-handler.md

```diff
@@ -0,0 +1,265 @@
+
+Resource handlers are also useful for building control panels that allow the user to write back to the data source. For example, you could add a resource handler to update the state of an IoT device.
+
+## Implement the resource handler interface
+
+To add a resource handler to your app plugin, you need to implement the `backend.CallResourceHandler` interface for your app struct.
+
+```go
+func (a *MyApp) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+    return sender.Send(&backend.CallResourceResponse{
+        Status: http.StatusOK,
+        Body: []byte("Hello, world!"),
+    })
+}
+```
+
+You can then access your resources through the following endpoint: `http://<GRAFANA_HOSTNAME>:<PORT>/api/plugins/<PLUGIN_ID>/resources`
+
+In this example code, `PLUGIN_ID` is the plugin identifier that uniquely identifies your app.
+
+## Add support for multiple routes
+
+To support multiple routes in your app plugin you have a couple of options depending on requirements and needs.
+If you only need basic support for a couple of different routes retrieving data you can use a switch with the `req.Path`:
+
+```go
+func (a *MyApp) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+	if req.Method != http.MethodGet {
+		return sender.Send(&backend.CallResourceResponse{
+			Status: http.StatusNotFound,
+		})
+	}
+
+	switch req.Path {
+	case "namespaces":
+		return sender.Send(&backend.CallResourceResponse{
+			Status: http.StatusOK,
+			Body:   []byte(`{ "namespaces": ["ns-1", "ns-2"] }`),
+		})
+	case "projects":
+		return sender.Send(&backend.CallResourceResponse{
+			Status: http.StatusOK,
+			Body:   []byte(`{ "projects": ["project-1", "project-2"] }`),
+		})
+	default:
+		return sender.Send(&backend.CallResourceResponse{
+			Status: http.StatusNotFound,
+		})
+	}
+}
+```
+
+Supporting additional routes and methods (GET, POST etc) pretty quickly gets cumbersome and creates code that might be hard to maintain, test and read. Here we recommend to use the more Go-agnostic approach for handling resources and use the regular [`http.Handler`](https://pkg.go.dev/net/http#Handler). You can do so by using a package provided by the [Grafana Plugin SDK for Go](../../key-concepts/backend-plugins/grafana-plugin-sdk-for-go) named [`httpadapter`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter). This package provides support for handling resource calls using an [`http.Handler`](https://pkg.go.dev/net/http#Handler).
+
+Using [`http.Handler`](https://pkg.go.dev/net/http#Handler) allows you to also use Go’s built-in router functionality called [`ServeMux`](https://pkg.go.dev/net/http#ServeMux) or your preferred HTTP router library (for example, [`gorilla/mux`](https://github.com/gorilla/mux)).
+
+:::note
```

Comment:
It'd much prefer this was the first content in the article instead of a side note. Using the ServeMux should be the "default" way to go about calling resources in  most cases.

What if move this on top explaining in the section `Implement the resource handler interface` that you have two options: Use ServeMux or manually implement the interface. Then you explain how to do it with ServeMux and finally how to do it using the call resource handler. e.g.:
---

# Implement the resource handler interface

To add a resource handler to your app plugin, you need to implement the `backend.CallResourceHandler` interface for your app struct.

There are two ways you can implement this in your plugin, using ServeMux or manually implementing it in your plugin:

## Using ServeMux

// this whole section

## Manually implementing `backend.CallResourceHandler` 

// the original part with and the multiple routes section go here